src/ucc/api/ucc.h

```diff
@@ -1097,22 +1131,113 @@ typedef struct ucc_ep_map_t {
+      * the number of outstanding calls to be posted.
+      */
     uint64_t                outstanding_colls;
+
+    /** @ref ucc_team_params.ep The endpoint is a non-negative unique integer identifying the
+      * participant in the collective. If ep is not set, and @ref ucc_team_params.oob is not set, the
+      * library generates the ep. The generated ep can be queried using the @ref
+      * ucc_team_get_attr interface.
+      */
     uint64_t                ep;
+
+    /** @ref ucc_team_params.ep_list The endpoint list provides the list of eps participating to
+     *  create the team.
+     */
     uint64_t                *ep_list;
+
+    /** @ref ucc_team_params.ep_range can be either contiguous or not
+     *  contiguous. It is a hint to the library.
+     */
     ucc_ep_range_type_t     ep_range;
+
+    /** @ref ucc_team_params.team_size The team size is the number of participants in the team. If
+      * @ref ucc_team_params.oob is provided, the team size and @ref
+      * ucc_oob_coll.n_oob_eps should be the same.
+      */
     uint64_t                team_size;
+
+    /**
+      * @ref ucc_team_params.sync_type The options for sync_type are provided by @ref
+      * ucc_coll_sync_type_t
+      */
     ucc_coll_sync_type_t    sync_type;
+
+   /** @ref ucc_team_params.oob The signature of the function is defined by @ref
+     * ucc_oob_coll_t
+     * . The oob is used for exchanging information between the team
+     * participants during team creation. The user is
+     * responsible for implementing the oob operation. The relation between @ref
+     * ucc_team_params.ep and @ref ucc_oob_coll.oob_ep is defined as below:
+     *
+     * - When both are not provided. The library is responsible for generating the ep,
+     * which can be then queried via the @ref ucc_team_get_attr interface. This
+     * requires, however,
+     * ucc_params_t ep_map to be set and context created by @ref ucc_oob_coll.
+     * The behavior is
+     * undefined, when neither @ref ucc_team_params.ep or @ref
+     * ucc_team_params.ep_map, or @ref ucc_team_params.oob is not set.
+     *
+     * - When @ref ucc_team_params.ep is provided and @ref ucc_team_params.oob is
+     * not provided. The “ep” is the unique integer
+     * for the participant.
+     *
+     * - When @ref ucc_oob_coll.oob_ep is provided and @ref ucc_team_params.ep
+     * is not provided. The “ep” will be
+     * equivalent to @ref ucc_oob_coll.oob_ep.
+     *
+     * - When both are provided, the @ref ucc_oob_coll.oob_ep and @ref
+     * ucc_team_params_t.ep should be same. Otherwise, it
+     * is undefined.
+     */
     ucc_team_oob_coll_t     oob;
+
+    /** @ref ucc_team_params.p2p_conn is a callback function for the gathering
+      * the point-to-point communication information.
+      */
     ucc_team_p2p_conn_t     p2p_conn;
+
+    /** @ref ucc_team_params.mem_params provides an ability to attach a buffer
```

Comment:
from the description of mem_params here several things are not clear:
1. What type/kind of memory can user pass? Which UCC_MEMORY_TYPEs (gpu/cpu) ? Should it be pinned?
2. What the library can do with that memory? - after team is created (user passed the memory into the ucc), does the ownership goes entirely to libucc? when is it returned to used, upon team destruction?
3. Should/can this memory be used as coll_args dst/src fields?

In general, an example of usage for 1-sided colls would be good.

Reply:
@vspetrov We are reworking on how to use this parameter. As a part of #309, we will propose the change. Let's discuss this in today's meeting.

Reply:
On this PR, I will not modify the params. The other PR (#310 ) will address any changes in this field, ok?